flink/README.md

```diff
@@ -96,7 +96,7 @@ If you want to specify the type by yourself, please use `Flink SQL`. It supports
 | tidb.replica-read.blacklist    | null          | Do not select TiKV store with given ip addresses. |
 | tidb.database.name             | null          | Database name. It is required for table factory, no need for catalog. |
 | tidb.table.name                | null          | Table name. It is required for table factory, no need for catalog. |
-| timestamp-format.${columnName} | null          | For each column, you could specify timestamp format in two cases: 1. TiDB `timestamp` is mapped to Flink `string`; 2. TiDB `varchar` is mapped to Flink `timestamp`. Format of timestamp may refer to `java.time.format.DateTimeFormatter`, like `yyyy-MM-dd HH:mm:ss.SSS`. It is optional for table factory, no need for catalog. |
+| tidb.timestamp-format.${columnName} | null          | For each column, you could specify timestamp format in two cases: 1. TiDB `timestamp` is mapped to Flink `string`; 2. TiDB `varchar` is mapped to Flink `timestamp`. Format of timestamp may refer to `java.time.format.DateTimeFormatter`, like `yyyy-MM-dd HH:mm:ss.SSS`. It is optional for table factory, no need for catalog. (You can still use the `timestamp-format.${columnName}` configuration item which will be officially removed in flink-1.14) |
```

Comment:
We should either leave the old config in documentation and mark it deprecated and only works in 1.11 and 1.12 or make sure 1.11 and 1.12 also works with both new and old config. 

Otherwise people might get confused that the new configuration doesn't work on 1.11 and 1.12.

My suggestion is adding compatibility patch to all the versions. Meanwhile make people clear in the document that 'tidb.timestamp-format' is the long term supported and 'timestamp-format' is going to be removed in next version.

Reply:
@Al-assad PTAL

Reply:
got it

Reply:
Hi @sunxiaoguang, thanks for your valuable suggestion, here are my relevant improvements:
* extracts and reuses the processing code of the timestamp-format configuration, which support for removing downward compatible behavior based on method parameters;
* add additional unit tests in versions 1.11, 1.12 to provide test coverage of `tidb.timestamp-format.`;
* modify the relevant instructions in the README again;